### PR TITLE
chain/querier: Refactor querier codebase

### LINKED ISSUE #1691: Modify querier type and keeper
We have our DB and event to expose state to external, but some function still be needed (get file from hash)

### PR DESCRIPTION
**Pull request by sorawit**
fixed #1691

### CODE REVIEW

File: chain/x/oracle/client/cli/query.go [hunk 1]
```diff
@@ -21,54 +23,81 @@ func GetQueryCmd(storeKey string, cdc *codec.Codec) *cobra.Command {
...
+			return cliCtx.PrintOutput(out)
+		},
+	}
+}
 
-			res, _, err := cliCtx.QueryWithData(
-				fmt.Sprintf("custom/%s/pending_request", queryRoute),
-				nil,
-			)
+// GetCmdDataSource implements the query data source command.
```
Comment (taobun) [inline chain/x/oracle/client/cli/query.go:71]: What input of this cli id or hash? We can explain or show an example in long description.
(https://github.com/cosmos/cosmos-sdk/blob/master/x/staking/client/cli/tx.go)

File: chain/x/oracle/types/querier.go [hunk 2]
```diff
@@ -1,111 +1,17 @@
...
-	QueryOracleScriptByID = "oracle_script"
-	QueryOracleScripts    = "oracle_scripts"
-	QueryRequestByID      = "request"
-	QueryRequests         = "requests"
-	QueryPending          = "pending_request"
-	QueryRequestNumber    = "request_number"
+	QueryParams        = "params"
+	QueryCounts        = "counts"
+	QueryData          = "data"
+	QueryDataSources   = "data_sources"
```
Comment (taobun) [inline chain/x/oracle/types/querier.go:8]: We want to query single data source?

File: chain/x/oracle/client/rest/rest.go [hunk 3]
```diff
@@ -9,17 +9,15 @@ import (
...
-	r.HandleFunc(fmt.Sprintf("/%s/oracle_script/{%s}", storeName, oracleScriptIDTag), getOracleScriptByIDHandler(cliCtx, storeName)).Methods("GET")
-	r.HandleFunc(fmt.Sprintf("/%s/oracle_scripts", storeName), getOracleScriptsHandler(cliCtx, storeName)).Methods("GET")
-	r.HandleFunc(fmt.Sprintf("/%s/request/{%s}", storeName, requestIDTag), getRequestByIDHandler(cliCtx, storeName)).Methods("GET")
-	r.HandleFunc(fmt.Sprintf("/%s/requests", storeName), getRequestsHandler(cliCtx, storeName)).Methods("GET")
-	r.HandleFunc(fmt.Sprintf("/%s/request_number", storeName), getRequestNumberHandler(cliCtx, storeName)).Methods("GET")
+	r.HandleFunc(fmt.Sprintf("/%s/params", storeName), getParamsHandler(cliCtx, storeName)).Methods("GET")
+	r.HandleFunc(fmt.Sprintf("/%s/counts", storeName), getCountsHandler(cliCtx, storeName)).Methods("GET")
+	r.HandleFunc(fmt.Sprintf("/%s/data/{%s}", storeName, dataHashTag), getDataByHashHandler(cliCtx, storeName)).Methods("GET")
+	r.HandleFunc(fmt.Sprintf("/%s/data_sources/{%s}", storeName, dataSourceIDTag), getDataSourceByIDHandler(cliCtx, storeName)).Methods("GET")
```
Comment (taobun) [inline chain/x/oracle/client/rest/rest.go:21]: data_source?

**Review by taobun — CHANGES_REQUESTED** [review]

File: chain/x/oracle/client/cli/query.go (hunk 1 above)
Comment (sorawit) [reply chain/x/oracle/client/cli/query.go:71]: I don't think we need long desc. I'll fix short desc to be more understandable.

File: chain/x/oracle/types/querier.go (hunk 2 above)
Comment (sorawit) [reply chain/x/oracle/types/querier.go:8]: Let's follow REST convention

**Review by taobun — APPROVED** [review]

**Comment by taobun** [comment]
@perimeko please review this pr

**Review by perimeko — APPROVED** [review]

**Pull request merged**